### PR TITLE
feat: add `titleTemplate` option (#22)

### LINKED ISSUE #22: [Feature Request] Handle titleTemplate like vue-meta
Hi,
It could be really cool to have a titleTemplate [like vue-meta](https://vue-meta.nuxtjs.org/api/#titletemplate), to be able to set a part of title for each page, and a global part.

For now, I've just made my own use head that just do that:
```ts
import { useHead as originalUseHead, HeadObject } from '@vueuse/head'

export const useHead = (head: HeadObject) => originalUseHead({
  ...head,
  title: `${head.title} - MyProject`
})

```

But it can't be customized in the app.

Thanks for your amazing work,
Mathieu.

### PR DESCRIPTION
**Pull request by BobbieGoede**
* Adds an options argument to `createHead`.
* Adds `titleTemplate` to `HeadObject`
* Closes #22 

The `titleTemplate` property has the type `MaybeRef<string> | ((chunk?: MaybeRef<string>) => string)`. 
When setting it to a string it accepts a template such as `%s | Website`  which turns into `My title | Website` when `title` is set to `My title`.

I haven't added any tests yet, will add tests unless more changes are required, please let me know!

### CODE REVIEW

**Comment by joshwcorbett** [comment]
Looking forward to this getting merged

**Comment by gilesbutler** [comment]
Are there any plans to merge this @egoist?

File: src/index.ts [hunk 1]
```diff
@@ -226,10 +262,11 @@ const updateElements = (
   )
 }
 
-export const createHead = () => {
+export const createHead = (options: HeadOptions = {}) => {
```
Comment (antfu) [inline src/index.ts:265]: I am a bit hesitant about this options approach. Couldn't we make it dynamic as an option of `useHead`?

Comment (innocenzi) [reply src/index.ts:265]: I took a spin at this in https://github.com/vueuse/head/pull/77 before realizing this PR existed. I can confirm this option is needed. In an Inertia app, the title template would exist in a `main.ts`, not in an SFC. And I also couldn't make SSR work properly without using `createHead`.

**Pull request merged**